packages/contract-utils/merkle-distributor/src/storage.rs

```diff
@@ -94,20 +92,27 @@ where
     ///
     /// * topics - `["set_claimed"]`
     /// * data - `[index: u32]`
+    ///
+    /// # Security Warning
+    ///
+    /// **IMPORTANT**: This function lacks authorization checks and should only
+    /// be used in admin functions that implement their own authorization logic.
     pub fn set_claimed(e: &Env, index: u32) {
         let key = MerkleDistributorStorageKey::Claimed(index);
         e.storage().persistent().set(&key, &true);
         emit_set_claimed(e, index.into());
     }
 
-    /// Verifies a Merkle proof for a node and marks its index as claimed if the
-    /// proof is valid.
+    /// Verifies a Merkle proof for a leaf and marks its index as claimed if the
+    /// proof is valid. Internally using [`Verifier::verify`] which assumes that
+    /// when the tree gets constructed, **commutative** hashing was used,
+    /// i.e.the leaves are **sorted**.
```

Comment:
I recall from the other issue #270, merkle tree hash implementation was relaxed in terms of removing the requirement of sorting. I think that was also for verification. Just wanted to double-check if everything is in order

Reply:
yes, both are present: `verify_and_set_claimed` for sorted trees and `verify_with_index_and_set_claimed` for unsorted